src/main/java/seedu/address/model/task/DistinctTaskList.java

```diff
@@ -43,6 +45,26 @@ public void addTask(Task taskAdded) {
         taskList.add(taskAdded);
     }
 
+    /**
+     * Replaces the task {@code target} in the list with {@code editedTask}.
+     * {@code target} must exist in the list.
+     * The task identity of {@code editedTask} must not be the same as another existing task in the list.
+     */
+    public void setTask(Task target, Task editedTask) {
+        requireAllNonNull(target, editedTask);
+
+        int index = taskList.indexOf(target);
+        if (index == -1) {
+            throw new TaskNotFoundException();
+        }
+
+        if (!target.isSameTask(editedTask) && contains(editedTask)) {
```

Comment:
Just want to clarify if target task is supposed to be different from editedTask to throw error or the same task to throw an error. From the code, it appears that if the two task are different then an error will be thrown. However, if this is changed to target.isSameTask, it will throw an error for the mark command. One way to resolve this is to remove the duplicate task exception since the task which is being changed should be the same. In addition, contains in this case should be target instead of editedTask as stated in javadocs

Reply:
setTask is meant to edit a task by replacing the target task with the editedTask, so these 2 tasks should have the same identities (module code and description). As such, it is supposed to throw error when the two tasks have different task identities (checked by isSameTask), which is also the expected result for markCommand. In addition, the editedTask should not have the same identity as any other tasks in the list as well, so this is checked by contains.

Reply:
I agree with what you have said but if that were the case, then maybe the exception thrown should not be DuplicateTaskException since DuplicateTaskException is supposed to be thrown for duplicate tasks being added rather than different tasks being added.

Reply:
I understand what you’re saying. There is a difference between 2 tasks having the same identity and 2 task objects being equal, but I feel that DuplicateTaskException should be thrown when either of them happens. The former might even be more significant because having the same identity already indicates that the tasks are same and hence, duplicated